idaes/generic_models/unit_models/column_models/column.py

```diff
@@ -0,0 +1,1144 @@
+                                      initialize=160,
+                                      doc='Enhancement factor')
+
+        # Heat transfer terms
+        self.heat_transfer_rate_vap = Var(self.flowsheet().time,
+                            self.vapor_phase.length_domain,
+                            domain=Reals,
+                            initialize=0.0,
+                            units=pyunits.J / (pyunits.s * pyunits.m),
+                            doc='Heat transfer rate in vapor phase')
+        
+        self.heat_transfer_rate_liq = Var(self.flowsheet().time,
+                            self.vapor_phase.length_domain,
+                            domain=Reals,
+                            initialize=0.0,
+                            units=pyunits.J / (pyunits.s * pyunits.m),
+                            doc='Heat transfer rate in liquid phase')
+
+        # =====================================================================
+        # Add performance equations
+
+        # Inter-facial Area model ([m2/m3]):
+
+        self.area_interfacial = Var(self.flowsheet().time,
+                                    self.vapor_phase.length_domain,
+                                    initialize=0.9,
+                                    doc='Specific inter-facial area')
+
+        # ---------------------------------------------------------------------
+        # Geometry constraints
+
+        # Column area [m2]
+        @self.Constraint(doc="Column cross-sectional area")
+        def column_cross_section_area(blk):
+            return blk.area_column == (
+                CONST.pi * 0.25 * (blk.diameter_column)**2)
+
+        # Area of control volume : vapor side and liquid side
+        control_volume_area_definition = ''' column_area * phase_holdup.
+        The void fraction of the vapor phase (volumetric vapor holdup) and that
+        of the liquid phase(volumetric liquid holdup) are
+        lumped into the definition of the cross-sectional area of the
+        vapor-side and liquid-side control volume respectively. Hence, the
+        cross-sectional area of the control volume changes with time and space.
+        '''
+
+        if self.config.dynamic:
+            @self.Constraint(self.flowsheet().time,
+                             self.vapor_phase.length_domain,
+                             doc=control_volume_area_definition)
+            def vapor_side_area(bk, t, x):
+                return bk.vapor_phase.area[t, x] == (
+                    bk.area_column * bk.holdup_vap[t, x])
+
+            @self.Constraint(self.flowsheet().time,
+                             self.liquid_phase.length_domain,
+                             doc=control_volume_area_definition)
+            def liquid_side_area(bk, t, x):
+                return bk.liquid_phase.area[t, x] == (
+                    bk.area_column * bk.holdup_liq[t, x])
+        else:
+            self.vapor_phase.area.fix(value(self.area_column))
```

Comment:
I think we should just write the constraints at all times - yes, fixing the variables makes the problem a bit less complicated and removes a couple of trivial constraints, but doing so adds a structural difference between dynamic and steady-state models. We have pre-solve tools that can clean up the trivial constraints if required.

Also, why are the phase areas fixed to the column area rather than `area*phase fraction`?